internal/discharger: change short-term stores

mhilton opened the pull request:
Change the types stored in the short-term data stores used in the login
protocol to store an Identites unmodifiable ProviderID rather that a
DischargeToken. DischargeTokens are now only created when they are
needed. This is a preliminary branch to support using ProviderIDs in the
discharge tokens.

Please note that, while this change is isolated to the discharge
package, some in-progress logins that are happening at the time of
upgrade to this code may fail. If logins do fail then they will work
if retryed immediately afterwards.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/45/

mhilton commented on charm/Makefile:
```diff
@@ -6,8 +6,20 @@ hooks/charmhelpers: charm-helpers.yaml
 lint:
-	flake8 hooks/config-changed hooks/db-relation-changed hooks/install.py hooks/jaascharm.py hooks/nrpe-relation-changed hooks/status.py hooks/stop hooks/update-status hooks/upgrade-charm hooks/website-relation-joined
-
+	@# ls -d hooks/* | grep -v -E '/(install|charmhelpers)$'
```
Is this line doing anything now?

rogpeppe replied:
It's just so the command is easily available to run to keep the lines below in sync.

mhilton commented on charm/metadata.yaml:
```diff
@@ -19,8 +19,10 @@ provides:
     interface: mongodb
+  postgres:
+    interface: pgsql
   # TODO allow postgres relation too?
```
TODO is TODONE

rogpeppe replied:
TODONE done

mhilton commented on charm/hooks/postgres-relation-changed:
```diff
@@ -0,0 +1,25 @@
+if __name__ == '__main__':
+    hookenv.log('db-relation-changed')
+    relation_changed()
+    jaascharm.update_status(failed_status=charm_status)
```
I'd be expecting charm_status to change too.

rogpeppe replied:
good point. done.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/46/

mhilton commented on charm/hooks/status.py:
```diff
@@ -6,10 +6,10 @@ def charm_status(msg):
-        # because the mongodb application at the other
+        # because the application at the other
         # end of the relation hasn't set its attributes yet.
         return 'waiting', 'waiting for db relation'
```
It might be worth checking these individually so that the message matches the ones in the relation-changed hooks?

rogpeppe replied:
good point. done.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/47/

rogpeppe commented:
:shipit:

jujugui commented:
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

mhilton committed:
internal/discharger: change short-term stores

Change the types stored in the short-term data stores used in the login
protocol to store an Identites unmodifiable ProviderID rather that a
DischargeToken. DischargeTokens are now only created when they are
needed. This is a preliminary branch to support using ProviderIDs in the
discharge tokens.

Please note that, while this change is isolated to the discharge
package, some in-progress logins that are happening at the time of
upgrade to this code may fail. If logins do fail then they will work
if retryed immediately afterwards.
